dataprofiler/labelers/data_processing.py

```diff
@@ -598,23 +600,38 @@ def process(self, data, labels=None, label_mapping=None, batch_size=32):
         Flatten batches of data
 
         :param data: List of strings to create embeddings for
-        :type data: list() of strings
+        :type data: numpy.ndarray
         :param labels: labels for each input character
-        :type labels: list of labels for each character in data
+        :type labels: numpy.ndarray
         :param label_mapping: maps labels to their encoded integers
         :type label_mapping: Union[None, dict]
         :param batch_size: Number of samples in the batch of data
         :type batch_size: int
         :return batch_data: A dict containing  samples of size batch_size
         :rtype batch_data: dicts
         """
-        if labels is not None and not label_mapping:
-            raise ValueError('If `labels` are specified, `label_mapping` must '
-                             'also be specified.')
+        if len(data.shape) > 1:
+            # Ignore case of 1 col DataFrame
+            if not (len(data.shape) == 2 and data.shape[1] == 1):
```

Comment:
it is possible this could be more than 2 dimensions.

I would suggest validating that only 1 value in shape is > 1 to account for all cases

Reply:
Good idea